src/main/java/school/lemon/changerequest/java/extendedinteger/ExtendedInteger.java

```diff
@@ -5,115 +5,141 @@
  */
 public class ExtendedInteger {
 
+    private int value;
+
     public ExtendedInteger(int value) {
-        //TODO: implement me
+        this.value = value;
     }
 
     /**
      * Check whether specified value is even
+     *
      * @param value to check
      * @return true if value is even, false - otherwise
      */
     public static boolean isEven(int value) {
-        //TODO: implement me
-        return false;
+        return value % 2 == 0;
     }
 
     /**
      * Check whether specified value is odd
+     *
      * @param value to check
      * @return true if value is odd, false - otherwise
      */
     public static boolean isOdd(int value) {
-        //TODO: implement me
-        return false;
+        return value % 2 != 0;
     }
 
     /**
      * Check whether specified value is prime
+     *
      * @param value to check
      * @return true if value is prime, false - otherwise
      */
     public static boolean isPrime(int value) {
-        //TODO: implement me
-        return false;
+        if (value < 2)
+            return false;
+        for (int i = 2; i * i <= value; i++)
+            if (value % i == 0)
+                return false;
+        return true;
     }
 
     /**
      * Parse specified char array and create instance of {@code ExtendedInteger}
+     *
      * @param value to parse
      * @return instance of {@code ExtendedInteger} or
      * null in case specified value is null or the value does not contain a parsable integer
      */
     public static ExtendedInteger parseInt(char[] value) {
-        //TODO: implement me
+        if (value == null || value.length == 0)
         return null;
+        String newValue = new String(value);
+        return ExtendedInteger.parseInt(newValue);
     }
 
     /**
      * Parse specified string and create instance of {@code ExtendedInteger}
+     *
      * @param value to parse
      * @return instance of {@code ExtendedInteger} or
      * null in case specified value is null or the value does not contain a parsable integer
      */
     public static ExtendedInteger parseInt(String value) {
-        //TODO: implement me
+        if (value == null | value.length() == 0)
         return null;
+        StringBuilder sb = new StringBuilder();
+        if (Character.isDigit(value.charAt(0)) || value.startsWith("-"))
+            sb = sb.append(value.charAt(0));
+        for (int i = 1; i < value.length(); i++) {
```

Comment:
As far as I see, your implementation will correctly process smth like: `G123` - where first character will be any symbol.